cluster-autoscaler/core/scaledown/actuation/delete_in_batch.go

```diff
@@ -67,59 +67,55 @@ func NewNodeDeletionBatcher(ctx *context.AutoscalingContext, csr *clusterstate.C
 	// If delete interval is 0, than instantly start node deletion.
 	if d.deleteInterval == 0 {
-		nodeGroup, err := deleteNodesFromCloudProvider(d.ctx, []*apiv1.Node{node})
-		if err != nil {
-			result := status.NodeDeleteResult{ResultType: status.NodeDeleteErrorFailedToDelete, Err: err}
-			CleanUpAndRecordFailedScaleDownEvent(d.ctx, node, nodeGroup.Id(), drain, d.nodeDeletionTracker, "", result)
-		} else {
-			RegisterAndRecordSuccessfulScaleDownEvent(d.ctx, d.clusterState, node, nodeGroup, drain, d.nodeDeletionTracker)
+		err := deleteNodesFromCloudProvider(d.ctx, nodes, nodeGroup)
+		for _, node := range nodes {
+			if err != nil {
+				result := status.NodeDeleteResult{ResultType: status.NodeDeleteErrorFailedToDelete, Err: err}
+				CleanUpAndRecordFailedScaleDownEvent(d.ctx, node, nodeGroup.Id(), drain, d.nodeDeletionTracker, "", result)
+			} else {
+				RegisterAndRecordSuccessfulScaleDownEvent(d.ctx, d.clusterState, node, nodeGroup, drain, d.nodeDeletionTracker)
+			}
 		}
-		return nil
-	}
-	nodeGroupId, first, err := d.addNodeToBucket(node, drain)
-	if err != nil {
-		return err
+		return
 	}
+	first := d.addNodesToBucket(nodes, nodeGroup, drain)
 	if first {
-		go func(nodeGroupId string) {
+		go func(nodeGroup cloudprovider.NodeGroup) {
 			time.Sleep(d.deleteInterval)
-			d.remove(nodeGroupId)
-		}(nodeGroupId)
+			d.executeForBucket(nodeGroup)
+		}(nodeGroup)
 	}
-	return nil
 }
 
 // AddToBucket adds node to delete candidates and return if it's a first node in the group.
-func (d *NodeDeletionBatcher) addNodeToBucket(node *apiv1.Node, drain bool) (string, bool, error) {
+func (d *NodeDeletionBatcher) addNodesToBucket(nodes []*apiv1.Node, nodeGroup cloudprovider.NodeGroup, drain bool) bool {
 	d.Lock()
 	defer d.Unlock()
-	nodeGroup, err := d.ctx.CloudProvider.NodeGroupForNode(node)
-	if err != nil {
-		return "", false, err
+	for _, node := range nodes {
+		d.drainedNodeDeletions[node.Name] = drain
 	}
-	d.drainedNodeDeletions[node.Name] = drain
 	val, ok := d.deletionsPerNodeGroup[nodeGroup.Id()]
 	if !ok || len(val) == 0 {
-		d.deletionsPerNodeGroup[nodeGroup.Id()] = []*apiv1.Node{node}
-		return nodeGroup.Id(), true, nil
+		d.deletionsPerNodeGroup[nodeGroup.Id()] = nodes
+		return true
 	}
-	d.deletionsPerNodeGroup[nodeGroup.Id()] = append(d.deletionsPerNodeGroup[nodeGroup.Id()], node)
-	return nodeGroup.Id(), false, nil
+	d.deletionsPerNodeGroup[nodeGroup.Id()] = append(d.deletionsPerNodeGroup[nodeGroup.Id()], nodes...)
+	return false
 }
 
-// remove delete nodes of a given nodeGroup, if successful, the deletion is recorded in CSR, and an event is emitted on the node.
-func (d *NodeDeletionBatcher) remove(nodeGroupId string) error {
+// executeForBucket deletes nodes of a given nodeGroup, if successful, the deletion is recorded in CSR, and an event is emitted on the node.
+func (d *NodeDeletionBatcher) executeForBucket(nodeGroup cloudprovider.NodeGroup) error {
```

Comment:
nit: I'm not sure if `executeForBucket` is more readable than `remove`. With `remove` one could deduce from the function name that something is being removed. With `executeForBucket` it's harder to understand what is about to be executed.

Reply:
done